index.html

```diff
@@ -1275,7 +1275,71 @@ <h2>The <dfn>InteractionOutput</dfn> interface</h2>
       </ol>
     </section>
   </section>
-  <section> <h3>Using {{InteractionInput}} and {{InteractionOutput}}</h3>
+  
+  <section data-dfn-for="ActionInteractionOutput">
+    <h2>The <dfn>ActionInteractionOutput</dfn> interface</h2>
+    <p>
+      Belongs to the <a>WoT Consumer</a> conformance class.
+      An {{ActionInteractionOutput}} object is always created by the implementations
+      and exposes the data returned from <a>WoT Interactions</a> to application
+      scripts.
+    </p>
+    <p>
+      This interface exposes an action status object. Its implementation
+      will allow cancelling asynchronous actions and report the status of a long running action.
+    </p>
+    <pre class="idl">
+      enum ActionStatus {
+        /* "pending", Profile has pending. Not sure what it would mean? */
+        "running",
+        "success", /* Profile uses completed? */
+        "error" /* Profile uses failed? */
+      };
+      
+      [SecureContext, Exposed=(Window,Worker)]
+      interface ActionInteractionOutput : InteractionOutput {
+        readonly attribute object? error;
+        Promise&lt;ActionStatus&gt; status();
+        Promise&lt;undefined&gt; cancel();
+      };
+    </pre>
+    <p>
+      The <dfn>error</dfn> property represents a possible error, initially `null`.
+    </p>
+    <p class="ednote" title="Should state be a function or an attribute only?">
```

Comment:
I started with a function but I start thinking an attribute should be sufficient..

Reply:
An action is supposed to be always polled? Or could a script subscribe just for changes in status?

Reply:
> Mhh, I tend to think now that a a function for reporting the status would be more powerful and useful. What do people think? Other ideas?

Hmm, if I see it correctly, we _could_ also treat the attribute as a getter and add more logic to that if needed, right?

> Or could a script subscribe just for changes in status?

If we wanted to that, what would be the best way to model that? Adding a way for registering a callback function?

Reply:
Mhh, maybe we should decide what we need/want from the status

* is it likely that requesting the status may take a while? (or put differently do we need a Promise)
* do we need a way to observe a status change (would be nice but adds complexity)

My take is that having a promise function for `status()` is fine while I am not 100% sure we should allow for listening status changes...

Are there opinions/proposals?